### DISCUSSION TITLE
Roadmap 2021 plan

### DISCUSSION THREAD

**Discussion by mhmttosun** [codex-team/editor.js#1898] [General]
Thanks for this awesome block style editor. It is really amazing work.
I am wondering if there is a roadmap for future development. I couldn't find one.

I am curios what core team plan/think about following issues and what core team plans for the future.

Collaboration support #908
Full featured Table support
Copy / Paste support  #729
Full featured Undo / Redo support

**Comment by neSpecc** [comment]
We were thinking about a collaborative editing feature at editor.js. At the moment we are designing schemas and documentation, making investigations and experiments about improving our data-model to make it possible. 

It will require lots of work with the editor core and APIs. And it will be the Editor.js 3.0. I can't say when it can be released because of our busyness. We will start working on it this year.

Full featured Undo/Redo will be implemented in 3.0 because it requires same improvements as collaborative editing.

What about the new Table tool — its already implemented and will be released in a 1—3 weeks. There are some optimisations left. You can follow a [PR ](https://github.com/editor-js/table/pull/67)

**Comment by denemier** [comment]
hey, a quick question about block based wysiwyg editors in general.
is it technically possible to develope image,media support inside block table? or does block based editors have a limit so  it cant have another block inside another?
and where can i learn how to develope block based editor? any source?

**Comment by brenojac** [comment]
> hey, a quick question about block based wysiwyg editors in general.
> is it technically possible to develope image,media support inside block table? or does block based editors have a limit so it cant have another block inside another?
> and where can i learn how to develope block based editor? any source?

Hello,

I've got the same question and wonder if, for example, will it be possible to insert an image into a table column and hyperlink it, for example. Thats a feature I used in other non-blockbased editors... editor.js is so great. I hope it evolves more and more in the upcoming future. Congrats on this amazing work.

**Comment by SalahAdDin** [comment]
How many of these goals did you reach?

**Comment by FarooqAlaulddin** [comment]
Hello - where you at with Full 'Undo / Redo support' feature? Would like to see where the team is going in terms of design.
